src/msgs/staking/execute/unstake.rs

```diff
@@ -1,14 +1,60 @@
 use super::*;
 
 #[cfg_attr(feature = "cosmwasm", cw_serde)]
-#[cfg_attr(not(feature = "cosmwasm"), derive(Serialize))]
+#[cfg_attr(not(feature = "cosmwasm"), derive(Serialize, Debug, PartialEq))]
 #[cfg_attr(not(feature = "cosmwasm"), serde(rename_all = "snake_case"))]
 pub struct Execute {
     pub public_key: String,
     pub proof:      String,
+    #[cfg_attr(not(feature = "cosmwasm"), serde(serialize_with = "crate::types::serialize_as_str"))]
     pub amount:     U128,
 }
 
+impl SignSelf for Execute {
+    type Extra = u128;
+
+    fn proof(&self) -> Result<Vec<u8>> {
+        Ok(hex::decode(&self.proof)?)
+    }
+
+    fn msg_hash(&self, chain_id: &str, contract_addr: &str, sequence: Self::Extra) -> Result<Hash> {
+        Ok(hash([
+            "unstake".as_bytes(),
+            #[cfg(not(feature = "cosmwasm"))]
+            &self.amount.to_be_bytes(),
+            #[cfg(feature = "cosmwasm")]
+            &self.amount.to_be_bytes(),
```

Comment:
Aren't both `&self.amount.to_be_bytes()` the same? (aswel as on other places)

Reply:
They are; I was scared Rust would be mad, though, cause the amount is different types in each situation, but I'll change it in a spot and run the tests to see if it gets mad.